client/src/pages/AddCourse/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useHistory } from "react-router-dom";
+import { UserContext } from "utils/auth";
+import { db } from "config/firebase";
+import {
+  Container,
+  Typography,
+  Button,
+  TextField,
+  Grid,
+} from "@material-ui/core";
+import Navbar from "components/Navbar";
+import { addCourse } from "utils/firestore";
+
+const AddCourse: React.FunctionComponent = () => {
+  const { user, userData, addCourseContext } = useContext(UserContext);
+  const history = useHistory();
+
+  interface CourseData {
+    id: string;
+    name: string;
+    description: string;
+    educator: string;
+  }
+
+  const DEFAULT_COURSE_DATA: CourseData = {
+    id: "",
+    name: "",
+    description: "",
+    educator: userData!.firstName + " " + userData!.lastName,
+  };
+  const [courseData, setCourseData] = useState<CourseData>(DEFAULT_COURSE_DATA);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setCourseData({
+      ...courseData,
+      // convert to boolean only for loginData.isAdmin
+      [e.target.name!]: e.target.value,
+    });
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    addCourseContext(courseData.id);
```

Comment:
Thoughts on this logic/ practicallity of updating context b4 db
Update courses string id array in context --> 
addCourseToCourses(courseData) -->
updateCoursesForUser(courses, uid) -->
history.push("/me");

Reply:
If an error occurs when updating db / the course isn't correctly added, would we have to update context again to reflect that? 
Leaning towards updating db --> updating context only because that would ensure context is consistent with db since it would only be updated after successful db calls.